src/lib/impl/SubscriberImpl.js

```diff
@@ -290,15 +327,109 @@ class SubscriberImpl extends EventEmitter {
+    let proto = resp[2][0];
+    if(proto === 'UDPROS' && this._udp){
+      this._handleUdpTopicRequestResponse(resp, nodeUri)
+    } else if (proto === 'TCPROS' && this._tcp){
+      this._handleTcpTopicRequestResponse(resp, nodeUri)
+    } else {
+      this._log.warn(`Publisher supports only ${proto} but is not enabled`)
+    }
+  }
 
-    this._log.debug('Topic request response: ' + JSON.stringify(resp));
+  _handleUdpTopicRequestResponse(resp, nodeUri){
 
-    // resp[2] has port and address for where to connect
+    if(this._subServer){
+      return
+    }
+    // Creating udp socket
+    const socket = UDPSocket.createSocket('udp4');
+
+
+    // cache client in "pending" map.
+    // It's not validated yet so we don't want it to show up as a client.
+    // Need to keep track of it in case we're shutdown before it can be validated.
+    //this._pendingPubClients[socket.nodeUri] = socket;
+
+    // create a one-time handler for the connection header
+    // if the connection is validated, we'll listen for more events
+    let hh = UdprosUtils.parseUdpRosHeader(resp[2][5])
+    socket.on('error', (err) => {
+      this._log.warn('Subscriber client socket %s on topic %s had error: %s',
+                     socket.name, this.getTopic(), err);
+      socket.close();
+      this._disconnectClient(socket.nodeUri);
+    });
+    // init empty msg
+    let msg = {
+      blkN: 0,
+      msgId: -1,
+      buffer: Buffer.alloc(0),
+      connectionId: -1
+    }
+    socket.on('message', (dgramMsg, rinfo) => {
+      let header = UdprosUtils.deserializeHeader(dgramMsg)
+      if(!header){
+        this._log.warn('Unable to parse packet\'s header for topic %s', this.getTopic())
+        return
+      }
+      // first dgram message
+      const { connectionId, opCode, blkN, msgId } = header
+      switch(opCode){
+        // DATA0
+        case 0:
+          // no chunk
+          if(blkN === 1){
+            this._handleMessage(dgramMsg.slice(12))
+          } else {
+            msg = {
+              blkN,
+              msgId,
+              buffer: dgramMsg.slice(12),
+              connectionId
+            }
+          }
+          break
+
+        // DATAN
+        case 1:
+          if(msgId === msg.msgId && connectionId === msg.connectionId){
+            let buffer = Buffer.from(dgramMsg.slice(8))
+            msg.buffer = Buffer.concat([msg.buffer, buffer])
+            // last chunk
+            if(msg.blkN -1 === header.blkN ){
+              this._handleMessage(Buffer.from(msg.buffer))
+            }
+          }
+          break
+
+        // PING
+        case 2:
+          break
+        
+        // ERR
+        case 3:
+          break
+      }
+    });
+
+    socket.on('listening', () => {
+      const address = socket.address();
+      this._log.debug(`UDP socket bound: ${address.address}:${address.port}, Topic:${this.getTopic()}`);
+      
+    });
+
+    this._subServer = socket
+  
+    socket.bind(this._port);
```

Comment:
Can this be moved into RosNode.js so that there's only one UDP server, mimicking `_setupTcprosServer`? Should be able to just pass 0 for port too in order to get a random available port.